### ISSUE TITLE
[Bug] Banners disappears when adding multiple images to my notes

### ISSUE THREAD

**Issue by albertoloscerritos** [HandaArchitect/obsidian-banner-snippet#14]
Hi, I think I found a bug that make that makes the banner disappear after adding multiple images in my notes. The bug is kind of difficult to reproduce, meaning I don't have the exact steps to reproduce it. But in summary after adding multiple embedded images to a note that has a banner and an image icon, the banners will eventually disappear, and it will come after deleting certain a number of images from the note. Important to note that the banner don't always come back after deleting the last pasted image, sometimes it may require us to delete 2 or 3 images before the banner comes back. I recorder a gif (it's like 1:30 min long) where you can see all the previously mentioned behaviors. (sorry for the low quality gif, had to compress it because of github's size limit)

![giphy](https://github.com/user-attachments/assets/b4119d45-0b76-4578-b83d-c3ddacda7201)

I am using obsidian 1.7.3. sandbox vault and windows 11. Btw I am not sure if it only happens when adding embedded images, but I have only observed this behavior when doing so. Thanks for your time

**Comment by HandaArchitect** [comment]
I have tried replicating this and made minor code changes. Once I have released 1.3.2, please give it another try. Hoping to release the new version by tomorrow. Just adding a few more code changes, looks like some themes are breaking the banner.

**Comment by albertoloscerritos** [comment]
I will try this one and let you know the results. One question, does it make any difference placing the banner and the icon lines at the beginning or at the end of the file? I prefer to not place those line at the star of my note since I feel like they get "in the way" of the actual notes. It would be great if we could place those in a property field, but I guess that is not posible with css. 

Btw, regarding what you said about certain themes breaking the banner, I am using AnuPpuccin theme in case that's relevant. Thanks.

**Comment by HandaArchitect** [comment]
> One question, does it make any difference placing the banner and the icon lines at the beginning or at the end of the file?

Yes, the placing of the banner image does matter, especially in reading mode.

Obsidian renders only certain blocks of text depending on the scroll position. When the banner image is placed at the bottom of the note and there are many blocks of text above the banner image, the banner image may not be rendered when scrolled at the top of the text.

Not much I can do at that point.
